x-pack/plugin/core/src/javaRestTest/java/org/elasticsearch/xpack/core/DataStreamRestIT.java

```diff
@@ -42,19 +43,30 @@ public void testDSXpackInfo() {
         assertTrue((boolean) dataStreams.get("enabled"));
     }
 
+    @SuppressWarnings("unchecked")
     public void testDSXpackUsage() throws Exception {
         Map<?, ?> dataStreams = (Map<?, ?>) getLocation("/_xpack/usage").get("data_streams");
         assertNotNull(dataStreams);
         assertTrue((boolean) dataStreams.get("available"));
         assertTrue((boolean) dataStreams.get("enabled"));
         assertThat(dataStreams.get("data_streams"), anyOf(equalTo(null), equalTo(0)));
-
+        assertThat(dataStreams, hasKey("failure_store"));
+        Map<String, Integer> failureStoreStats = (Map<String, Integer>) dataStreams.get("failure_store");
+        assertThat(failureStoreStats.get("enabled"), anyOf(equalTo(null), equalTo(0)));
+        assertThat(failureStoreStats.get("failure_indices_count"), anyOf(equalTo(null), equalTo(0)));
+        assertBusy(() -> {
+            Map<?, ?> logsTemplate = (Map<?, ?>) ((List<?>) getLocation("/_index_template/logs").get("index_templates")).get(0);
+            assertThat(logsTemplate, notNullValue());
+            assertThat(logsTemplate.get("name"), equalTo("logs"));
+            assertThat(((Map<?, ?>) logsTemplate.get("index_template")).get("data_stream"), notNullValue());
+        });
         assertBusy(() -> {
             Map<?, ?> logsTemplate = (Map<?, ?>) ((List<?>) getLocation("/_index_template/logs").get("index_templates")).get(0);
             assertThat(logsTemplate, notNullValue());
             assertThat(logsTemplate.get("name"), equalTo("logs"));
             assertThat(((Map<?, ?>) logsTemplate.get("index_template")).get("data_stream"), notNullValue());
         });
```

Comment:
Am I missing something or did you perhaps duplicate this `assertBusy` accidentally?

Reply:
You are right :)